nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -747,15 +748,29 @@ def _collector(self, buffer: List) -> Callable[[Module, Tensor, Tensor], None]:
         buffer.append(0)
 
         def collect_activation(_module: Module, _input: Tensor, output: Tensor):
-            activation = self._activation_trans(output)
+            if isinstance(_module.module, (torch.nn.Linear, torch.nn.Bilinear)):
+                batch_nums = math.prod(output.shape[:-1])
+                batch_dims = [_ for _ in range(len(output.shape[:-1]))]
+            elif isinstance(_module.module, (torch.nn.Conv1d, torch.nn.ConvTranspose1d)):
+                batch_nums = math.prod(output.shape[:-2])
+                batch_dims = [_ for _ in range(len(output.shape[:-2]))]
+            elif isinstance(_module.module, (torch.nn.Conv2d, torch.nn.ConvTranspose2d)):
+                batch_nums = math.prod(output.shape[:-3])
+                batch_dims = [_ for _ in range(len(output.shape[:-3]))]
+            elif isinstance(_module.module, (torch.nn.Conv3d, torch.nn.ConvTranspose3d)):
+                batch_nums = math.prod(output.shape[:-4])
+                batch_dims = [_ for _ in range(len(output.shape[:-4]))]
+            else:
+                raise TypeError(f'Found unsupported module type in activation based pruner: {_module.__class__.__name__}')
```

Comment:
how about put this part into a utility function?

Reply:
good suggestion, moved